lib/src/stream_extensions.dart

```diff
@@ -54,4 +54,17 @@ extension StreamExtensions<T> on Stream<T> {
     });
     return completer.future;
   }
+
+  /// Returns a view of this stream that eagerly buffers all events until
+  /// `listen()` is called.
+  Stream<T> buffer() {
```

Comment:
Consider adding parameters like `([int? maxBuffer])`, which makes the buffering pause after `maxBuffer` events if set. Maybe `([int? maxBuffer, int Function(T)? weight])`, so you can assign a "size" to each event, e.g., the length of lists, and count that towards the maximum buffering size.

Reply:
Let's leave this for later if the need arises—I think the main use of this will be to buffer _all_ events, so I'm not sure having buffer size controls are likely to be valuable enough to add a positional parameter.

Reply:
I am worried about having a function which can cause unlimited memory bloat. I'd really like to force the user to *consider* the problem.

At least put something in the documentation about needing to listen to the stream. Even it it's only `.listen(null).cancel()`, it's needed to stop the underlying stream.